tests/unit/httpd_test.cc

```diff
@@ -1016,6 +1016,54 @@ static future<> test_basic_content(bool streamed, bool chunked_reply) {
                 BOOST_REQUIRE_EQUAL(body, sstring("12345 78901\t34521345"));
             }
 
+            {
+                fmt::print("Request with content-length body\n");
+                auto req = http::request::make("GET", "test", "/test");
+                req.write_body("txt", 12, [] (output_stream<char>&& out) {
+                    return seastar::async([out = std::move(out)] () mutable {
+                        out.write(sstring("1234567890")).get();
+                        out.write(sstring("ABCDEFGHIJ")).get();
+                        out.flush().get();
+                        out.close().get();
+                    });
+                });
+                auto resp = conn.make_request(std::move(req)).get0();
+                BOOST_REQUIRE_EQUAL(resp._status, http::reply::status_type::ok);
+                if (!chunked_reply) {
+                    BOOST_REQUIRE_EQUAL(resp.content_length, 12);
+                }
+                auto in = conn.in(resp);
+                sstring body = util::read_entire_stream_contiguous(in).get0();
+                BOOST_REQUIRE_EQUAL(body, sstring("1234567890AB"));
```

Comment:
I don't understand what I'm seeing here. If the body writer writes **fewer** than 12 characters, this will be an error (and you have a test for it). But here the body writer writes **too many** characters, 20 instead of 12, and you silently truncate it to 20? Why does this make sense? Isn't this a bug in the body writer, exactly like having too few characters is a bug? Why shouldn't this case cause an exception as well?

Reply:
Yes, if the body writer writes more than the requested amount of bytes, the tail it silently cut. This is what this test tests. I'm fine with changing it to be exception

Reply:
Yes, it seemed to me that the body writer should receive the exception when trying to write more than the allowed amount, as this is a bug in the code that writes the response - just like writing too little. I commented about this here, but obviously my thinking is that the code - not just the tests - should be changed.
But if you think I'm wrong and it's useful for the body writer to be allowed to write too much and be silently trucated, maybe you can explain to me why you think that's useful.